stress/shell/src/main/java/alluxio/stress/cli/suite/Suite.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.stress.cli.suite;
+
+import alluxio.stress.Summary;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.ParametersDelegate;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Base class for all benchmark suites.
+ *
+ * @param <T> the summary class
+ */
+public abstract class Suite<T extends Summary> {
+  @ParametersDelegate
+  protected BaseParameters mBaseParameters = new BaseParameters();
+
+  private static final class BaseParameters {
+    static final String CLUSTER_FLAG = "--cluster";
+
+    @Parameter(names = {CLUSTER_FLAG},
+        description = "If true, runs the benchmark via the job service cluster. Otherwise, runs "
+            + "locally.")
+    public boolean mCluster = false;
+
+    @Parameter(names = {"--java-opt"},
+        description = "The java options to add to the command line to for the task. This can be "
+            + "repeated. The options must be quoted and prefixed with a space, to avoid getting "
+            + "passed to the JVM. For example: --java-opt \" -Xmx4g\" --java-opt \" -Xms2g\"")
```

Comment:
similar comment about having to _always_ add a space. I think we can do that ourselves?

Reply:
If we are passing these on to another program, then we will get these values values as strings in the java program beforehand. What I'm suggesting is that if every property is supposed to be quoted and have a space before it, why don't we make it easier to just pass the property and do this in the code:

```java
mJavaOpts = mJavaOpts.stream()
  .map(property -> String.format("\" %s\"", property))
  .collect(Collectors.toList()) 
```